src/fileio/fs_utils.cpp

```diff
@@ -272,6 +272,39 @@ EXPORT bool create_directory(const std::string& path) {
   return false;
 }
 
+EXPORT bool create_directory_or_throw(const std::string& path) {
+  bool result = create_directory(path);
+
+  // this function throws if the directory still doesn't exist
+  // at that location
+  file_status status = get_file_status(path);
+  switch (status) {
+    case file_status::MISSING:
```

Comment:
Case readonly?

Reply:
That isn't one of the cases in `file_status`.